kcbq-api/src/main/java/com/wepay/kafka/connect/bigquery/api/SchemaRetriever.java

```diff
@@ -16,22 +18,24 @@
    * {@link org.apache.kafka.connect.sink.SinkConnector#start(Map)} method.
    * @param properties The configuration settings of the connector.
    */
-  public void configure(Map<String, String> properties);
+  void configure(Map<String, String> properties);
 
   /**
    * Retrieve the most current schema for the given topic.
    * @param table The table that will be created.
-   * @param topic The topic to retrieve a schema for.
+   * @param topicAndRecordName The topic and an optional record name to retrieve a schema for.
```

Comment:
Record name is optional. I did not want to end up with methods that have too many arguments and overloads. 

It's been a while since I commercially worked with Java (I use Scala on my daily basis) and AFAICS passing parameters of type `Optional<T>` is not recommended (as one of the Java core team members, it's not aimed to be a counterpart of `Maybe` monad from Haskell or Scala's `Option`) which is reflected in my IDE's default warnings.

Passing nulls in place of a newly added optional parameters does not look good to me, hence plain `String` has been replaced with a dedicated type with an optional record type (which returns an `Optional<String>` and enforces its API users to handle the case with record type being absent. And that's the use case recommended by Java team members and authorities). 

Another motivation was to improve type safety, make method signatures more clear and avoid having some _util_ class with a static method that creates a subject from two strings.